src/wallet/wallet.cpp

```diff
@@ -392,10 +393,11 @@ void CWallet::UpgradeKeyMetadata()
             // Write meta to wallet
             CPubKey pubkey;
             if (GetPubKey(meta_pair.first, pubkey)) {
-                WriteKeyMetadata(meta, pubkey, true);
+                batch.WriteKeyMetadata(meta, pubkey, true);
             }
         }
     }
+    batch.TxnCommit(); //commit before setting the flag
```

Comment:
I used the explicit commit (rather then the RAII destructor) to avoid refolding the code.